src/overlays/actors/ovl_En_Daiku_Kakariko/z_en_daiku_kakariko.c

```diff
@@ -43,29 +58,464 @@ static ColliderCylinderInit D_809E54B0 = {
     },
     { 18, 66, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku_Kakariko/func_809E4320.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku_Kakariko/EnDaikuKakariko_Init.s")
+static CollisionCheckInfoInit2 sColChkInit = { 0, 0, 0, 0, 0xFF };
+
+static DamageTable sDamageTable = { 0 };
+
+static struct_D_80AA1678 sAnimations[] = {
+    { 0x06001AB0, 1.0f, 2, -7.0f }, { 0x06007DE0, 1.0f, 0, -7.0f }, { 0x0600885C, 1.0f, 0, -7.0f },
+    { 0x06000C44, 1.0f, 0, -7.0f }, { 0x06000600, 1.0f, 0, -7.0f }, { 0x06008164, 1.0f, 0, -7.0f },
+};
+
+void EnDaikuKakariko_SetAnimFromIndex(EnDaikuKakariko* this, s32 animIndex, s32* currentAnimIndex) {
+    f32 transitionRate;
+
+    if ((*currentAnimIndex < 0) || (animIndex == *currentAnimIndex)) {
+        transitionRate = 0.0f;
+    } else {
+        transitionRate = sAnimations[animIndex].transitionRate;
+    }
+
+    Animation_Change(&this->skelAnime, sAnimations[animIndex].animation, 1.0f, 0.0f,
+                     Animation_GetLastFrame(sAnimations[animIndex].animation), sAnimations[animIndex].mode,
+                     transitionRate);
```

Comment:
transistionrate is now called morphFrames. there may be some other animation related changes you need to make for temp vars